content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md

```diff
@@ -0,0 +1,243 @@
+---
+title: "Kubernetes Response Engine, Part 8: Falcosidekick + Flux v2"
+date: 2021-06-21
+author: Batuhan Apaydın
+slug: falcosidekick-reponse-engine-part-8-fluxv2
+---
+
+> *This blog post is part of a series of articles about how to create a `Kubernetes` response engine with `Falco`, `Falcosidekick` and a `FaaS`.*
+> 
+> See other posts:
+> * [Kubernetes Response Engine, Part 1 : Falcosidekick + Kubeless]({{< ref "/blog/falcosidekick-reponse-engine-part-1-kubeless" >}})
+> * [Kubernetes Response Engine, Part 2 : Falcosidekick + OpenFaas]({{< ref "/blog/falcosidekick-reponse-engine-part-2-openfass" >}})
+> * [Kubernetes Response Engine, Part 3 : Falcosidekick + Knative]({{< ref "/blog/falcosidekick-reponse-engine-part-3-knative" >}})
+> * [Kubernetes Response Engine, Part 4 : Falcosidekick + Tekton]({{< ref "/blog/falcosidekick-reponse-engine-part-4-tekton" >}})
+> * [Kubernetes Response Engine, Part 5 : Falcosidekick + Argo]({{< ref "/blog/falcosidekick-reponse-engine-part-5-argo" >}})
+
+----
+
+Today, we will set up another KRE (Kubernetes Response Engine) based on Flux v2. If you don't know about Flux v2, let me explain it in a few words. Flux is a tool for keeping Kubernetes clusters in sync with sources of configuration (like Git repositories), and automating updates to configuration when there is new code to deploy.
+
+To learn more about Flux v2, see [Flux Documentation](https://fluxcd.io/docs/).
+
+You might be thinking, Flux v2 sounds like a GitOps tool, really it is btw, how does it fit to the KRE setup, because in the previous blog posts related to set up KRE, we used some kind of eventing systems such as `Knative Eventing`, `Tekton Triggers`, `Argo Events`, etc to forward events from one point to another but Flux v2 is not supporting the same kind of eventing systems by it's design because you know it is just a GitOps tool, it basically watches GitHub repositories for changes, keeps the current state and the desired state always in sync, and it is doing that on top of Kubernetes with the make use of a bunch of CRs (Custom Resources). In order to set up KRE with Flux v2, we will create a small go project that listens events and updates the GitHub repository which Flux v2 monitors for changes to change the desired state for example get the event of pwned pod, then change it's replicas to a zero within the deployment.yaml file.
+
+Given below is a reference architecture of what's being explained in this blog.
```

Comment:
The reference architecture given below illustrates the content of this blog.

Reply:
thank you so much for fixing the issues, done.